### PR TITLE
MEN-4919: auth set status change validation

### PR DESCRIPTION
**Pull request by 0lmi**
https://tracker.mender.io/browse/MEN-4919

### CODE REVIEW

File: devauth/devauth.go [hunk 1]
```diff
@@ -829,24 +829,27 @@ func (d *DevAuth) AcceptDeviceAuth(ctx context.Context, device_id string, auth_i
 		return errors.Wrap(err, "db get auth set error")
 	}
 
+	// device authentication set already accepted, nothing to do here
+	if aset.Status == model.DevStatusAccepted {
+		l.Debugf("Device %s already accepted", device_id)
+		return nil
+	}
+	// device authentication set can be accepted only from 'pending' or 'rejected' statuses
+	if aset.Status != model.DevStatusRejected && aset.Status != model.DevStatusPending {
```
Comment (tranchitella) [inline devauth/devauth.go:838]: nitpick: can be `else if`

File: devauth/devauth.go [hunk 2]
```diff
@@ -951,7 +954,19 @@ func (d *DevAuth) setAuthSetStatus(
...
-	err := d.cacheDeleteToken(ctx, device_id)
+	aset, err := d.db.GetAuthSetById(ctx, auth_id)
+	if err != nil {
+		if err == store.ErrAuthSetNotFound {
+			return err
+		}
+		return errors.Wrap(err, "db get auth set error")
+	}
+	// device authentication set can be rejected only from 'accepted' or 'pending' statuses
+	if aset.Status != model.DevStatusPending && aset.Status != model.DevStatusAccepted {
```
Comment (tranchitella) [inline devauth/devauth.go:965]: nitpick: `else if`

File: devauth/devauth.go [hunk 3]
```diff
@@ -960,6 +975,17 @@ func (d *DevAuth) RejectDeviceAuth(ctx context.Context, device_id string, auth_i
...
 func (d *DevAuth) ResetDeviceAuth(ctx context.Context, device_id string, auth_id string) error {
+	aset, err := d.db.GetAuthSetById(ctx, auth_id)
+	if err != nil {
+		if err == store.ErrAuthSetNotFound {
+			return err
+		}
+		return errors.Wrap(err, "db get auth set error")
+	}
+	// preauthorized auth set should not go into pending state
+	if aset.Status == model.DevStatusPreauth {
```
Comment (tranchitella) [inline devauth/devauth.go:986]: nitpick: `else if`

**Review by tranchitella — COMMENTED** [review]
@0lmi looks good to me, let's run the tests!

**Comment by 0lmi** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
Custom QA pipeline: https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/356528224

**Review by tranchitella — APPROVED** [review]

**Comment by 0lmi** [comment]
@mender-test-bot start pipeline

**Review by tranchitella — APPROVED** [review]

**Pull request merged**